src/Rector/Contrib/CodeQuality/UnnecessaryTernaryExpressionRector.php

```diff
@@ -0,0 +1,117 @@
+use PhpParser\Node\Expr\BinaryOp\Smaller;
+use PhpParser\Node\Expr\BinaryOp\SmallerOrEqual;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\Ternary;
+use PhpParser\Node\Identifier;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+final class UnnecessaryTernaryExpressionRector extends AbstractRector
+{
+    /**
+     * @var string
+     */
+    private $ifValue;
+
+    /**
+     * @var string
+     */
+    private $elseValue;
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition(
+            'Remove unnecessary ternary expressions.',
+            [new CodeSample('$foo === $bar ? true : false;', '$foo === $bar;')]
+        );
+    }
+
+    public function isCandidate(Node $node): bool
+    {
+        if (! $node instanceof Ternary) {
+            return false;
+        }
+
+        /** @var Ternary $ternaryExpression */
+        $ternaryExpression = $node;
+
+        if (! $ternaryExpression->if instanceof Expr) {
+            return false;
+        }
+
+        $condition = $ternaryExpression->cond;
+        if (! $condition instanceof BinaryOp) {
+            return false;
+        }
+
+        $ifExpression = $ternaryExpression->if;
+        $elseExpression = $ternaryExpression->else;
+
+        if (! $ifExpression instanceof ConstFetch
+            || ! $elseExpression instanceof ConstFetch
+        ) {
+            return false;
+        }
+
+        /** @var Identifier $ifExpressionName */
+        $ifExpressionName = $ifExpression->name;
+        /** @var Identifier $elseExpressionName */
+        $elseExpressionName = $elseExpression->name;
+
+        $this->ifValue = $ifExpressionName->toLowerString();
+        $this->elseValue = $elseExpressionName->toLowerString();
+
+        return ! in_array('null', [$this->ifValue, $this->elseValue], true);
+    }
+
+    /**
+     * @param Ternary $ternaryNode
+     */
+    public function refactor(Node $ternaryNode): ?Node
+    {
+        /** @var BinaryOp */
+        $binaryOpOperation = $ternaryNode->cond;
+
+        if ($this->ifValue === 'true' && $this->elseValue === 'false') {
+            $ternaryNode = $binaryOpOperation;
+        } else {
+            $ternaryNode = $this->fixBinaryOperation($binaryOpOperation);
+        }
+
+        return $ternaryNode;
+    }
+
+    private function fixBinaryOperation(BinaryOp $operation): BinaryOp
+    {
+        $operandsMap = [
+            '===' => NotIdentical::class,
+            '!==' => Identical::class,
+            '==' => NotEqual::class,
+            '!=' => Equal::class,
+            '<>' => Equal::class,
+            '>' => Smaller::class,
+            '<' => Greater::class,
+            '>=' => SmallerOrEqual::class,
+            '<=' => GreaterOrEqual::class,
+        ];
+
+        $operand = $operation->getOperatorSigil();
+        $binaryOpClassName = $operandsMap[$operand];
```

Comment:
What about `<=>`?

Reply:
How would you refactory it?

Reply:
I'd add test first

Reply:
Any updated on this or ETA?

Reply:
I couldn’t understand your point. `<=>` can’t be refactored.

Reply:
Maybe I don't understand this part of code. Could you explain it?

Reply:
Maybe I don't understand this part of code. Could you explain it?